exim/init.sls

```diff
@@ -21,35 +21,42 @@ exim_install_packages:
 {% set relay_domains = salt['pillar.get']('exim:config:relay_domains', '') %} 
 {% set relay_nets = salt['pillar.get']('exim:config:relay_nets', '') %} 
 {% set smarthost = salt['pillar.get']('exim:config:smarthost', '') %} 
+{% set cfilemode = salt['pillar.get']('exim:config:cfilemode', '644') %} 
 
 {{ map.config_dir }}/{{ map.config_file }}:
   file.managed:
     - contents: |
-        dc_use_split_config='{{ use_split_config }}'
         dc_eximconfig_configtype='{{ configtype }}'
+        dc_other_hostnames='{{ other_hostnames }}'
+        dc_local_interfaces='{{ local_interfaces }}'
+        dc_relay_domains='{{ relay_domains }}'
+        dc_relay_nets='{{ relay_nets }}'
+        dc_use_split_config='{{ use_split_config }}'
         dc_hide_mailname='{{ hide_mailname }}'
         ue4c_keepcomments='{{ ue4c_keepcomments }}'
         dc_localdelivery='{{ localdelivery }}'
-        dc_local_interfaces='{{ local_interfaces }}'
         dc_minimaldns='{{ minimaldns }}'
-        dc_other_hostnames='{{ other_hostnames }}'
         dc_readhost='{{ readhost }}'
-        dc_relay_domains='{{ relay_domains }}'
-        dc_relay_nets='{{ relay_nets }}'
         dc_smarthost='{{ smarthost }}'
+        CFILEMODE='{{ cfilemode }}'
     - watch_in:
       - service: {{ map.service }}
 
+{% if salt['pillar.get']('exim:files') %}
 {% for dir in map.sub_dirs %}
   {% for file in salt['pillar.get']('exim:files:' + dir, {}) %}
 {{ map.config_dir }}/conf.d/{{ dir }}/{{ file }}:
   file.managed:
-    - source: salt://exim/{{ file }}
+    - contents_pillar: exim:files:{{ dir }}:{{ file }}
     - watch_in:
       - service: {{ map.service }}
   {% endfor %}
 {% endfor %}
+{% endif %}
+
 
 exim_service:
   service.running:
     - name: {{ map.service }}
+
```

Comment:
Minor style issue: better to avoid trailing newlines. You may want to add something right after the last line of actual code if jump at the bottom of the file.